js/hooks/useNfc.ts

```diff
@@ -8,20 +8,36 @@ export type NfcResult =
   | { status: "cancelled" }
   | { status: "error"; error: unknown };
 
-export const useNfc = (cancel: AbortController): NfcResult => {
+export const useNfc = (): {
+  result: NfcResult;
+  abortController: AbortController | null;
+} => {
   const supported = nfcSupported();
 
   const [result, setResult] = useState<NfcResult>({
     status: supported ? "reading" : "nfcUnsupported",
   });
 
+  const [abortController, setAbortController] =
+    useState<AbortController | null>(null);
```

Comment:
Should this be a ref? Does its value changing need to result in a re-render?

Reply:
Since the `abortController` is part of the return value of the hook, we need it to be updated as part of the React state if we want changes to the `abortController` to actually trigger a rerender and therefore a new call to `useNfc` and a new return value for the caller. If it was purely internal to this function I think a ref might make sense (or even just a local variable).